### DISCUSSION TITLE
pc.cond works only in localhost...

### DISCUSSION THREAD

**Discussion by cheuora** [reflex-dev/reflex#863] [Q&A] [answered]
I have made a test page with pc.cond() and tested in my pc. There are no problems running in my pc with 'localhost:3000'. But If I move my sources to the ubuntu server(I access it with public IP), pc.cond() doesn't work. Ubuntu server shows static pages well. But when I click a button which trigger pc.cond(), nothing happen. 

Do you have any experience like this?

**Comment by cheuora** [comment, marked as answer]
I found the reason of this problem. 

my pcconfig.py file was wrong. 

```
import pynecone as pc

config = pc.Config(
    app_name="pyne",
    api_url = 'http://<my ubuntu>:3000',
    bun_path="$HOME/.bun/bin/bun",
    db_url="sqlite:///pynecone.db",
    env=pc.Env.DEV,
)
```
I have changed `api_url` port num to 8000 and pc.cond() starts work. 

Do you have any idea what relationship between pc.cond() and port 8000?

**Reply by picklelo** [reply]
Without using the correct backend port, your app can't get the correct user state. In fact, none of the events in your app would work.

**Reply by picklelo** [reply]
The 3000 port is for the frontend, while the 8000 port is for the backend. We use websockets to communicate between the two. We can improve the docs to explain this better.
